forms/gridfield/GridFieldExportButton.php

```diff
@@ -115,12 +115,14 @@ protected function getExportColumnsForGridField(GridField $gridField) {
 	 * Generate export fields for CSV.
 	 *
 	 * @param GridField $gridField
-	 * @return array
+	 * @return string
 	 */
 	public function generateExportFileData($gridField) {
 		$separator = $this->csvSeparator;
 		$csvColumns = $this->getExportColumnsForGridField($gridField);
-		$fileData = '';
+
+		$stream = fopen('php://memory', 'w');
```

Comment:
Also see http://www.ozonesolutions.com/programming/2011/10/use-fputcsv-to-make-csv-output-to-browser-or-variable/